gobblin-data-management/src/main/java/org/apache/gobblin/data/management/conversion/hive/source/HiveAvroToOrcSource.java

```diff
@@ -36,8 +43,36 @@
     if (!state.contains(HiveDatasetFinder.HIVE_DATASET_CONFIG_PREFIX_KEY)) {
       state.setProp(HiveDatasetFinder.HIVE_DATASET_CONFIG_PREFIX_KEY, "hive.conversion.avro");
     }
+    this.lineageInfo = LineageInfo.getLineageInfo(state.getBroker());
+
+    List<WorkUnit> workunits = super.getWorkunits(state);
+    for (WorkUnit workUnit : workunits) {
+      if (shouldSetLineageInfo(workUnit)) {
+        setSourceLineageInfo(workUnit, (ConvertibleHiveDataset) ((HiveWorkUnit) workUnit).getHiveDataset(),
+            this.lineageInfo);
+      }
+    }
+    return workunits;
+  }
 
-    return super.getWorkunits(state);
+  @VisibleForTesting
+  public static void setSourceLineageInfo(WorkUnit workUnit, ConvertibleHiveDataset hiveDataset,
```

Comment:
Remove the `static` so that subclass add its own information. For test, we can initialize a `HiveAvroToOrc` object.